### ISSUE TITLE
Tags

### ISSUE THREAD

**Issue by xyproto** [plougher/squashfs-tools#61]
Hi, could you please make a `3.1` tag for the last version of squashfs that supports the 2.4 kernel? Thanks!

**Comment by plougher** [comment]
Hi,

I have tagged both the 3.1 release and the 3.2-r2 release, and i have also updated the README to better reflect the position regarding 2.4 kernel support.

"kernel-2.4: If you still need Squashfs support in the 2.4 kernel
then use the squashfs 3.2-r2 release.  This is the last release
that has a Mksquashfs which generates filesystems mountable with
Squashfs patched 2.4 kernels.  This release has patches for 2.4
kernels (but they have not been updated since the 3.1 release)."

Strictly speaking, the 3.1 release is the last release that had updated patches for 2.4 kernels.   But in practice the 3.2-r2 release is the last release that has a Mksquashfs which generates filesystems mountable with 2.4 patched kernels (a MAJOR of 3, and a MINOR of 0).  The distinction is perhaps important.

The 3.1 release still had a non-multithreaded Mksquashfs.  This was highly reliable, but slow because it only used one core.  It did also have an experimental multithreaded Mksquashfs (par_mksquashfs), but this was not default as it was still in development, and not 100% reliable.

The 3.2-r2 release has the multithreaded Mksquashfs as default, which is now reliable, and it is a lot faster than the previous single-threaded Mksquashfs.

I would recommend you use release 3.2-r2 as it has the multithreaded Mksquashfs.  If you prefer to use the 3.1 release, then I would recommend that you do not use the experimental par_mksquashfs.

**Comment by xyproto** [comment]
Great, thank you!

**Comment by plougher** [comment]
One last comment.  If you use the 3.2-r2 Mksquashfs then add the option "--no-exports".

From release 3.2 onwards Squashfs filesystems were exportable by default via NFS.  This required an additional set of information in the filesystem.

This extra information is ignored by Squashfs  3.1 patched kernels (i.e. any 2.4 patched kernels).   But as this extra information is ignored, it is unnecessary baggage, and you should remove it using the -no-exports option.

**Comment by xyproto** [comment]
Thank you, this is really helpful. I use mksqashfs to create an image for an old hardware platform.

**Comment by plougher** [comment]
No problem, I try to be helpful with this "ancient" history, that probably few people now remember.

Unfortunately, I have discovered I have made an embarrassing mistake with the tagging.  I have tagged the 3.3 release with the 3.2 tag :-(  Unfortunately, I'm dealing here with pre-git CVS commits imported into the git repository, and it is very difficult to match it up against releases, as the CVS history is not very good.

I have added an additional tag 3.2-r2 which is the correct tag to use for the 3.2-r2 release.
